### PR TITLE
Add component tests for filtering by topics

### PR DESCRIPTION
**Pull request by RMPall**
### What

Add component tests to cover scenarios for filtering by topics. 

### How to review

Please check if the changes make sense. 

### Who can review

Anyone except me.

### CODE REVIEW

File: features/search-with-topics.feature [hunk 1]
```diff
@@ -0,0 +1,52 @@
+Feature: Search endpoint should return data for requested search parameter
+    Scenario: When Searching for a canonical topic and a sub topic topics I get multiple docs
+         Given elasticsearch returns multiple items in search response with topics filter
+         When I GET "/search?topics=123,004"
+         Then the HTTP status code should be "200"
+         And the response header "Content-Type" should be "application/json;charset=utf-8"
+         And the response body is the same as the json in "./features/testdata/expected_multiple_topics_search_result.json"
+
+    Scenario: When Searching for multiple sub topics I get multiple docs
```
Comment (nshumoogum) [inline features/search-with-topics.feature:9]:
```suggestion
    Scenario: When filtering search results with multiple valid topics, I get multiple docs returned. Both documents have at least one value in "topics" matching 1 of the topic filters
```

File: features/search-with-topics.feature [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+         Given elasticsearch returns multiple items in search response with topics filter
+         When I GET "/search?topics=002,004"
+         Then the HTTP status code should be "200"
+         And the response header "Content-Type" should be "application/json;charset=utf-8"
+         And the response body is the same as the json in "./features/testdata/expected_multiple_topics_search_result.json"
+
+    Scenario: When Searching for multiple canonical topics I get multiple docs
```
Comment (nshumoogum) [inline features/search-with-topics.feature:16]:
```suggestion
    Scenario: When filtering search results with multiple valid topics, I get multiple docs returned. Both documents "canonical_topic" matching 1 of the topic filters
```

File: features/search-with-topics.feature [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+         Given elasticsearch returns multiple items in search response with topics filter
+         When I GET "/search?topics=123,456"
+         Then the HTTP status code should be "200"
+         And the response header "Content-Type" should be "application/json;charset=utf-8"
+         And the response body is the same as the json in "./features/testdata/expected_multiple_topics_search_result.json"
+
+    Scenario: When Searching for single canonical topic I get single doc
```
Comment (nshumoogum) [inline features/search-with-topics.feature:23]:
```suggestion
    Scenario: When filtering search results with a single valid topic, I only get docs returned that have "canonical_topic" field matching the topic filter
```

File: features/search-with-topics.feature [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+         Given elasticsearch returns one item in search response with topics filter
+         When I GET "/search?topics=123"
+         Then the HTTP status code should be "200"
+         And the response header "Content-Type" should be "application/json;charset=utf-8"
+         And the response body is the same as the json in "./features/testdata/expected_single_topic_search_result.json"
+
+    Scenario: When Searching for single sub topic I get single doc
```
Comment (nshumoogum) [inline features/search-with-topics.feature:30]:
```suggestion
    Scenario: When filtering search results with a single valid topic, I only get docs returned that contain a value in "topics" field matching the topic filter
```

File: features/search-with-topics.feature [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+         Given elasticsearch returns one item in search response with topics filter
+         When I GET "/search?topics=001"
+         Then the HTTP status code should be "200"
+         And the response header "Content-Type" should be "application/json;charset=utf-8"
+         And the response body is the same as the json in "./features/testdata/expected_single_topic_search_result.json"
+
+    Scenario: When Searching for multiple canonical topics, one that exists and another that does not exist I get single doc
```
Comment (nshumoogum) [inline features/search-with-topics.feature:37]:
```suggestion
    Scenario: When filtering search results with multiple topics, 1 invalid and 1 valid topic. I only get docs returned that have "canonical_topic" field matching the valid topic filter
```

File: features/search-with-topics.feature [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+         Given elasticsearch returns one item in search response with topics filter
+         When I GET "/search?topics=123,7000"
+         Then the HTTP status code should be "200"
+         And the response header "Content-Type" should be "application/json;charset=utf-8"
+         And the response body is the same as the json in "./features/testdata/expected_single_topic_search_result.json"
+
+    Scenario: When Searching for multiple sub topics, one that exists and another that does not exist I get single doc
```
Comment (nshumoogum) [inline features/search-with-topics.feature:44]:
```suggestion
    Scenario: When filtering search results with multiple topics, 1 invalid and 1 valid topic. I only get docs returned that contain a value in "topics" field matching the valid topic filter
```

**Review by nshumoogum — CHANGES_REQUESTED** [review]

File: features/search-with-topics.feature [hunk 7]
```diff
@@ -0,0 +1,52 @@
+Feature: Search endpoint should return data for requested search parameter
+    Scenario: When Searching for a canonical topic and a sub topic topics I get multiple docs
```
[the remark this answers is not in the document]
Comment (RMPall) [reply features/search-with-topics.feature]: Fixed in update 1.

File: features/search-with-topics.feature (hunk 1 above)
Comment (RMPall) [reply features/search-with-topics.feature]: Fixed in update 1.

File: features/search-with-topics.feature (hunk 2 above)
Comment (RMPall) [reply features/search-with-topics.feature]: Fixed in update 1.

File: features/search-with-topics.feature (hunk 3 above)
Comment (RMPall) [reply features/search-with-topics.feature]: Fixed in update 1.

File: features/search-with-topics.feature (hunk 4 above)
Comment (RMPall) [reply features/search-with-topics.feature]: Fixed in update 1.

File: features/search-with-topics.feature (hunk 6 above)
Comment (RMPall) [reply features/search-with-topics.feature]: Fixed in update 1.

**Review by nshumoogum — APPROVED** [review]

**Pull request merged**